auctioneer/src/llm_api.rs

```diff
@@ -0,0 +1,69 @@
+
+use kinode_process_lib::{
+    http::{send_request, send_request_await_response, Method},
+    our_capabilities, spawn, Address, OnExit, ProcessId, Request, println
+};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::{path::PathBuf, str::FromStr};
+
+use crate::llm_types::openai::{
+    ChatParams, ChatRequest, LLMRequest, LLMResponse, Message as OpenaiMessage,
+};
+use crate::prompts::create_prompt;
+
+pub fn init_openai_pkg(our: Address) -> anyhow::Result<Address> {
+    let openai_pkg_path = format!("{}/pkg/openai.wasm", our.package_id());
+    let our_caps = our_capabilities();
+    let http_client = ProcessId::from_str("http_client:distro:sys").unwrap();
+
+    let process_id = spawn(
+        None,
+        &openai_pkg_path,
+        OnExit::None,
+        our_caps,
+        vec![http_client],
+        false,
+    )?;
+
+    let worker_address = Address {
+        node: our.node.clone(),
+        process: process_id.clone(),
+    };
+
+    Ok(worker_address)
+}
+
+// TODO: Zen: This can later be used in llm_types, as part of the spec
+pub fn chat(text: &str, openai_key: &str, llm_worker: &Address) -> anyhow::Result<String> {
+    let msg = OpenaiMessage {
+        content: create_prompt(text),
+        role: "user".into(),
+    };
+
+    let chat_params = ChatParams {
+        model: "gpt-3.5-turbo".into(),
```

Comment:
these are currently all hardcoded, maybe the llm_api helpers could look a bit like some of our process_lib helpers?

So that init_openai_pkg (also call it spawn as that's what it does), would return a struct that stores it's model, key, temperature and max_tokens. Then Api.chat() can utilize self.model and so on.